onnxruntime/test/providers/cpu/tensor/transpose_test.cc

```diff
@@ -246,6 +246,39 @@ TEST(TransposeOpTest, ThreeDimSuffix) {
   TransposeTest(input_shape, input_vals, &perm, expected_shape, expected_vals, false);  //TensorRT: illegal error
 }
 
+TEST(TransposeOpTest, TransposeReshape) {
```

Comment:
Can we add test(s) for the kind of inputs that would have caused an issue previously  (i.e.) the reason why the PR as reverted ?

Reply:
Done. I checked this test failed with the previous version.

Reply:
Sorry - Did this specific test fail with the old version ? It seems to be the same as the old checked-in test....

Reply:
Yes it fails with the old version. The test I mention is here. https://github.com/microsoft/onnxruntime/pull/5916/files#diff-fc7da1e84aadf77e977b26bbf56319b4cf1ffb9ae7fbfdc539ab84cb2e971582R517 It creates a buffer bigger than the expected output and fills it with a constant. After the function is called, I checks the constant did not change outside the expected output.